staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/cbor.go

```diff
@@ -0,0 +1,304 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cbor
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/recognizer"
+	util "k8s.io/apimachinery/pkg/util/runtime"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+var encMode cbor.EncMode = func() cbor.EncMode {
+	encMode, err := cbor.EncOptions{
+		// Map keys need to be sorted to have deterministic output, and this is the order
```

Comment:
We only want to do this for storage into etcd, right?  Not for encoding from the kube-apiserver to REST clients.

Reply:
That's right. Since determinism is de facto part of the contract of Encode, the plan is to expose a second method for callers that don't need determinism, and explicitly use that second method in those cases.